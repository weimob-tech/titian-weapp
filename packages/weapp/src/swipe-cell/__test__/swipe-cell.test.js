const simulate = require('miniprogram-simulate');
const { getComponent, getComponentId } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('swipe-cell');
    expect(comp).toMatchSnapshot();
  });

  it('should basic', (done) => {
    const swipeCell = getComponentId('swipe-cell');
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'ti-swipe-cell': swipeCell
        },
        template: `
          <ti-swipe-cell id='swiper-cell' visible='{{visible}}' left-width='{{100}}' right-width='{{100}}'>
            <view slot='left'>
              <text>left</text>
            </view>
            <view>center</view>
            <view slot='right'>
              <text>right</text>
            </view>
          </ti-swipe-cell>
        `,
        data: {
          visible: false
        }
      })
    );
    comp.attach(document.createElement('parent-wrapper'));
    Promise.resolve()
      .then(
        () =>
          new Promise((resolve) => {
            setTimeout(() => {
              resolve();
            }, 1000);
          })
      )
      .then(() => {
        expect(comp.toJSON()).toMatchSnapshot();
        comp.setData({ visible: true });
      })
      .then(() => {
        expect(comp.toJSON()).toMatchSnapshot();
        comp.setData({ visible: false });
      })
      .then(() => {
        const swipeCellComp = comp.querySelector('#swiper-cell');
        const chooseImg = swipeCellComp.querySelector('.titian-swipe-cell');
        chooseImg.dispatchEvent('touchstart');
        chooseImg.dispatchEvent('touchmove');
        chooseImg.dispatchEvent('touchend');
        return new Promise((resolve) => {
          setTimeout(() => {
            resolve();
          }, 1000);
        });
      })
      .then(() => {
        comp.detach();
        done();
      });
  });
});
