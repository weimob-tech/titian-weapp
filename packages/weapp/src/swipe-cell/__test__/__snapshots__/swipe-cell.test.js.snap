// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should basic 1`] = `
<main>
  <main
    id="swiper-cell"
  >
    <wx-view
      class="ext-class titian-swipe-cell"
      data-key="cell"
      catch:tap="onClick"
      bind:touchcancel="onTouchcancel"
      bind:touchend="onTouchEnd"
      capture-bind:touchmove="onTouchMove"
      bind:touchstart="onTouchStart"
    >
      <wx-view
        class="titian-swipe-cell-wrap"
        style="
        -webkit-transform: translate3d(0rpx, 0, 0);
        -webkit-transition: transform .6s cubic-bezier(0.18, 0.89, 0.32, 1);
        transform: translate3d(0rpx, 0, 0);
        transition: transform .6s cubic-bezier(0.18, 0.89, 0.32, 1);
      "
      >
        <wx-view
          class="left-class titian-swipe-cell-left"
          data-key="left"
          catch:tap="onClick"
        >
          <wx-view
            slot="left"
          >
            <wx-text>
              left
            </wx-text>
          </wx-view>
        </wx-view>
        <wx-view
          class="titian-swipe-cell-content"
        >
          <wx-view>
            center
          </wx-view>
        </wx-view>
        <wx-view
          class="right-class titian-swipe-cell-right"
          data-key="right"
          catch:tap="onClick"
        >
          <wx-view
            slot="right"
          >
            <wx-text>
              right
            </wx-text>
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
  </main>
</main>
`;

exports[`test component should basic 2`] = `
<main>
  <main
    id="swiper-cell"
  >
    <wx-view
      class="ext-class titian-swipe-cell"
      data-key="cell"
      catch:tap="onClick"
      bind:touchcancel="onTouchcancel"
      bind:touchend="onTouchEnd"
      capture-bind:touchmove="onTouchMove"
      bind:touchstart="onTouchStart"
    >
      <wx-view
        class="titian-swipe-cell-wrap"
        style="
        -webkit-transform: translate3d(-100rpx, 0, 0);
        -webkit-transition: transform .6s cubic-bezier(0.18, 0.89, 0.32, 1);
        transform: translate3d(-100rpx, 0, 0);
        transition: transform .6s cubic-bezier(0.18, 0.89, 0.32, 1);
      "
      >
        <wx-view
          class="left-class titian-swipe-cell-left"
          data-key="left"
          catch:tap="onClick"
        >
          <wx-view
            slot="left"
          >
            <wx-text>
              left
            </wx-text>
          </wx-view>
        </wx-view>
        <wx-view
          class="titian-swipe-cell-content"
        >
          <wx-view>
            center
          </wx-view>
        </wx-view>
        <wx-view
          class="right-class titian-swipe-cell-right"
          data-key="right"
          catch:tap="onClick"
        >
          <wx-view
            slot="right"
          >
            <wx-text>
              right
            </wx-text>
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
  </main>
</main>
`;

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class titian-swipe-cell"
    data-key="cell"
    catch:tap="onClick"
    bind:touchcancel="onTouchcancel"
    bind:touchend="onTouchEnd"
    capture-bind:touchmove="onTouchMove"
    bind:touchstart="onTouchStart"
  >
    <wx-view
      class="titian-swipe-cell-wrap"
      style=""
    >
      <wx-view
        class="left-class titian-swipe-cell-left"
        data-key="left"
        catch:tap="onClick"
      />
      <wx-view
        class="titian-swipe-cell-content"
      />
      <wx-view
        class="right-class titian-swipe-cell-right"
        data-key="right"
        catch:tap="onClick"
      />
    </wx-view>
  </wx-view>
</main>
`;
